arm-network/2017-06-01/swagger/azureWAN.json

```diff
@@ -0,0 +1,2812 @@
+                    },
+                    {
+                        "name": "VirtualWANName",
+                        "in": "path",
+                        "required": true,
+                        "type": "string",
+                        "description": "The name of the VirtualWAN being retrieved."
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Request successful. Returns the details of the VirtualWAN retrieved.",
+                        "schema": {
+                            "$ref": "#/definitions/VirtualWAN"
+                        }
+                    },
+                    "default": {
+                        "description": "*** Error Response: ***\n\n * 404 VirtualWAN doesn't exist."
+                    }
+                }
+            },
+            "put": {
+                "operationId": "VirtualWANs_CreateOrUpdate",
+                "x-ms-examples": {
+                    "VirtualWANCreate": { "$ref": "../examples/VirtualWANPut.json" }
+                },
+                "description": "Creates a VirtualWAN resource if it doesn't exist else updates the existing VirtualWAN.",
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceGroupParameter"
+                    },
+                    {
+                        "name": "VirtualWANName",
+                        "in": "path",
+                        "required": true,
+                        "type": "string",
+                        "description": "The name of the VirtualWAN being created or updated."
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "name": "parameters",
+                        "in": "body",
+                        "required": true,
+                        "schema": {
+                            "$ref": "#/definitions/VirtualWAN"
+                        },
+                        "description": "Parameters supplied to create or update VirtualWAN."
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Request successful. Returns the details of the VirtualWAN created or updated.",
+                        "schema": {
+                            "$ref": "#/definitions/VirtualWAN"
+                        }
+                    },
+                    "202": {
```

Comment:
I know in NRP, the 202 async create pattern is widely used. But we recommend using the 201 + provisioning State + Azure-AsyncOperation header format for PUT/PATCH. Any specific reason why you use the 202 and cant use 201?